editor/store/effects/utils.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { select, subscribe } from '@wordpress/data';
+
+/**
+ * Waits for the resolution of a selector before returning the selector's value.
+ *
+ * @param {string} namespace    Store namespace.
+ * @param {string} selectorName Selector name.
+ * @param {Array} args          Selector args.
+ *
+ * @return {Promise} Selector result.
+ */
+export function resolveSelector( namespace, selectorName, ...args ) {
+	return new Promise( ( resolve ) => {
+		const hasFinished = () => select( 'core/data' ).hasFinishedResolution( namespace, selectorName, args );
+		const getResult = () => select( namespace )[ selectorName ].apply( null, args );
+
+		const unsubscribe = subscribe( () => {
+			if ( hasFinished() ) {
```

Comment:
This is where it's not working at the moment, I'm not certain why but `hasFinished` is never true. 

* Try to edit a post and click "save draft", nothing happens because it's stuck here.

cc @aduth

Reply:
Fixed by https://github.com/aduth/equivalent-key-map/pull/1